tests/server/watch/leader_watch_test.go

```diff
@@ -95,6 +95,6 @@ func TestWatcherCompacted(t *testing.T) {
 	err = pd2.Run()
 	re.NoError(err)
 	testutil.Eventually(re, func() bool {
-		return pd2.GetLeader().GetName() == pd1.GetConfig().Name
+		return cluster.WaitLeader() != "" && pd2.GetLeader().GetName() == pd1.GetConfig().Name
```

Comment:
This is more robust because it can handle the case of `nil`.

```suggestion
		return len(cluster.WaitLeader()) > 0 && pd2.GetLeader().GetName() == pd1.GetConfig().Name
```